### PR TITLE
unify network handlers

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@hzxuzhonghu PTAL as discussed in another PR

File: pilot/pkg/model/network.go [hunk 1]
```diff
@@ -79,20 +80,55 @@ func NewNetworkManager(env *Environment, xdsUpdater XDSUpdater) (*NetworkManager
 }
 
 func (mgr *NetworkManager) reloadAndPush() {
+	changed := mgr.reloadNetworkEndpoints()
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/network.go:83]: I think you should keep `reloadNetworkEndpoints` as another network handler and only register to `NetworksWatcher`

Comment (ramaraochavali) [reply pilot/pkg/model/network.go]: SGTM. Done

File: pilot/pkg/model/network.go [hunk 2]
```diff
@@ -71,28 +72,67 @@ func NewNetworkManager(env *Environment, xdsUpdater XDSUpdater) (*NetworkManager
...
+	mgr.mu.Unlock()
 
 	if changed && mgr.xdsUpdater != nil {
 		log.Infof("gateways changed, triggering push")
 		mgr.xdsUpdater.ConfigUpdate(&PushRequest{Full: true, Reason: []TriggerReason{NetworksTrigger}})
 	}
 }
 
+// reloadNetworkEndpoints reloads NetworkEndpoints and triggers a push if they change.
+func (mgr *NetworkManager) reloadNetworkEndpoints() {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/network.go:101]: sorry one more comment:  Missing from the original pr, seems this should be protected with EnableHCMInternalNetworks flag

Comment (ramaraochavali) [reply pilot/pkg/model/network.go:103]: Ah..Good catch

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**